Source/WebKit/Shared/curl/WebCoreArgumentCodersCurl.serialization.in

```diff
@@ -0,0 +1,37 @@
+# Copyright (C) 2024 Sony Interactive Entertainment Inc.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1.  Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+# 2.  Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS'' AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#if USE(CURL)
+
+[CreateUsing=tryMakeProxySettings] class WebCore::CurlProxySettings {
+    WebCore::CurlProxySettings::Mode mode();
```

Comment:
The new code is a bit less efficient than the old one. Ideally we'd send a `std::variant<Mode, std::pair<String, String>>` or something similar so we don't have to IPC 3 things when the mode is not custom.

Reply:
Is that more efficient though? The `Mode` is an `uint8_t` and if there's no proxy then the `String`s should be empty.

Reply:
Strings would be null but would still need to be serialized (as 2 booleans that are false).

Reply:
Likely not a massive issue but that's the approach we've taken in many places and it would match the previous serialization logic more closely.